sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/AmqpSendLink.java

```diff
@@ -24,16 +26,50 @@
      */
     Mono<Void> send(Message message);
 
+    /**
+     * Batches list of messages given into a single proton-j message that is sent down the wire.
+     *
+     * @param messages The list of messages to send to the service.
+     * @param deliveryState to be sent along with message.
+     *
+     * @return A Mono that completes when all the batched messages are successfully transmitted to message broker.
+     * @throws AmqpException if the serialized contents of {@code messageBatch} exceed the link's capacity for a single
+     * message.
+     */
+    Mono<Void> send(List<Message> messages, DeliveryState deliveryState);
```

Comment:
Is this used in production? iirc, we only send a single message, or an amqp message as a batch.

Reply:
sender client does reference this https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java#L266